.github/workflows/ci_tests_dev.yaml

```diff
@@ -87,11 +87,13 @@ jobs:
       # Install dependencies from conda-forge
       - name: Install dependencies
         run: |
-          mamba install ninja cmake libblas libcblas liblapack fftw gdal geopandas \
-                        ghostscript libnetcdf hdf5 zlib curl pcre make dvc
-          pip install --pre numpy pandas xarray netCDF4 packaging \
-                            ipython pytest-cov pytest-mpl pytest>=6.0 sphinx-gallery \
-                            tomli
+          mamba install python=${{ matrix.python-version }} \
+                        ninja cmake libblas libcblas liblapack fftw libgdal \
+                        geopandas ghostscript libnetcdf hdf5 zlib curl pcre make
```

Comment:
Do you think we should install geopandas using `pip` instead?

Reply:
Ideally yes, and I did try it (see https://github.com/GenericMappingTools/pygmt/pull/1577/commits/69d4e6716b964c96fd4b158fe6ac986bbdbb9300), but `pip install geopandas` doesn't work on Python 3.10 because the `fiona` dependency doesn't have Python 3.10 wheels yet and the install is not functional:

```
E           ImportError: the 'to_file' method requires the 'fiona' package, but it is not installed or does not import correctly.
E           Importing fiona resulted in: /usr/share/miniconda3/envs/pygmt/lib/python3.10/site-packages/fiona/ogrext.cpython-310-x86_64-linux-gnu.so: undefined symbol: _PyGen_Send
```

So I had to revert back to using `conda install geopandas` in https://github.com/GenericMappingTools/pygmt/pull/1577/commits/0604c950fac47b00cb7d6fcfc0fc15ef7e3a8498.

Reply:
FYI, Python 3.10 support in `fiona` is being worked on at https://github.com/Toblerity/Fiona/pull/1049. Seems like the Python 3.10 wheels might take a while to come out, so probably shouldn't wait.

Reply:
If we're waiting for the wheels to come out, why shouldn't we wait to upgrade to Python 3.10?

Reply:
> If we're waiting for the wheels to come out, why shouldn't we wait to upgrade to Python 3.10?

If I'm reading your sentence correctly, no, we don't have to wait for the `fiona` Python 3.10 wheels to come out. The current state on the main branch is that `fiona` is installed indirectly via `conda install geopandas`. @seisman's suggestion was that we use `pip install --pre geopandas` instead, but that doesn't work because `fiona` doesn't have Python 3.10 wheels.

Since geopandas/fiona is an optional dependency for PyGMT, I think we can just push ahead with this PR as is (i.e. stick with `conda install geopandas`), but happy to wait too if there's a good reason to.

Reply:
It seems the new fiona release will take at least another month (https://github.com/Toblerity/Fiona/issues/1053). I think we don't need to wait for it.